### PR TITLE
Chapter on Calico

### PR DESCRIPTION
**Pull request by nikipat**
*Issue #, if available:* #23 

*Description of changes:* Added a chapter on Calico

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/calico/stars_policy_demo/create_resources.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Lastly, [create](https://docs.projectcalico.org/v3.1/getting-started/kubernetes/tutorials/stars-policy/manifests/04-client.yaml) a client namespace, and a client service for a replication controller:
+
+```
+kubectl apply -f https://docs.projectcalico.org/v3.1/getting-started/kubernetes/tutorials/stars-policy/manifests/04-client.yaml
+```
+Check their status, and wait for all the pods to reach the Running status:
+
+```
+$ kubectl get pods --all-namespaces
+NAMESPACE       NAME                                                  READY   STATUS    RESTARTS   AGE
```
Comment (brentley) [inline content/calico/stars_policy_demo/create_resources.md]: The output needs to be separated from the command so it's not also copy/pasted. I think there are other examples of "Your output should look similar to this:" type of wording.

File: content/calico/stars_policy_demo/create_resources.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
+---
+title: "Create Resources"
+date: 2018-08-07T08:30:11-07:00
+weight: 1
+---
+
+Before creating network polices, let's create the required resources.
+
+Create a namespace called [stars](https://docs.projectcalico.org/v3.2/getting-started/kubernetes/tutorials/stars-policy/manifests/00-namespace.yaml). We will create frontend and backend services in this namespace in later steps
```
Comment (dalbhanj) [inline content/calico/stars_policy_demo/create_resources.md]: I'd like to modify this experience to --> here is yaml for the namespace. During testing next week, we will be targeting these yaml files and pose the audience with a challenge question and force them to edit these files. Don't forget to link to k8s docs on how to create namespace

File: content/calico/stars_policy_demo/create_resources.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+kube-system     calico-typha-horizontal-autoscaler-78f747b679-kzzwq   1/1     Running   0          28m
+kube-system     kube-dns-7cc87d595-bd9hq                              3/3     Running   0          1h
+kube-system     kube-proxy-lp4vw                                      1/1     Running   0          50m
+kube-system     kube-proxy-rfljb                                      1/1     Running   0          50m
+kube-system     kube-proxy-wzlqg                                      1/1     Running   0          50m
+management-ui   management-ui-wzvz4                                   1/1     Running   0          24m
+stars           backend-tkjrx                                         1/1     Running   0          24m
+stars           frontend-q4r84                                        1/1     Running   0          24m
+```
+
```
Comment (dalbhanj) [inline content/calico/stars_policy_demo/create_resources.md]: Can you summarize all the resources created and their namespaces

File: content/calico/stars_policy_demo/default_policy.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
+---
+title: "Default Pod-to-Pod Communication"
+date: 2018-08-07T08:30:11-07:00
+weight: 2
+---
+In Kubernetes, the **pods by default can communicate with other pods**, regardless of which host they land on. Every pod gets its own IP address so you do not need to explicitly create links between pods. This is demonstrated by the management UI
+
+You can access it by going to one of your worker nodes' IP address and specifying port :30002 at the end (for example: 52.12.161.128:30002)
+
+30002 is the port on each node's IP where the service is exposed.
```
Comment (dalbhanj) [inline content/calico/stars_policy_demo/default_policy.md]: I think this should say, because we specified NodePort while deploying our services, you can reach this service using worker node IP:nodeport

File: content/calico/stars_policy_demo/default_policy.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  namespace: management-ui 
+spec:
+  type: NodePort
+  ports:
+  - port: 9001 
+    targetPort: 9001
+    nodePort: 30002
+```
+{{% notice tip %}}
+Remember to open port 30002 in your Worker nodes' Security Group.
```
Comment (dalbhanj) [inline content/calico/stars_policy_demo/default_policy.md]: Offer link to Security Group documentation. Also, it would be nice if you provide direct link to SG console using search term (see previous chapters)

File: content/calico/stars_policy_demo/directional_traffic.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+And allow traffic from the client namespace to the frontend service:
+
+```
+kubectl apply -f https://docs.projectcalico.org/v3.1/getting-started/kubernetes/tutorials/stars-policy/policies/frontend-policy.yaml
+```
+Upon refreshing your browser, you should be able to see the network policies in action:
+
+![directional traffic](/images/calico-client-f-b-access.png)
+
+Let's have a look at the backend-policy. Its spec has a podSelector that selects all pods with the label **role:backend**, and allows ingress from all pods that have the label **role:frontend** and on TCP port **6379**
```
Comment (dalbhanj) [inline content/calico/stars_policy_demo/directional_traffic.md]: I would also add that the traffic is allowed in one direction and specific based on port number (6379 in this case)

File: content/calico/stars_policy_demo/create_resources.md (hunk 2 above)
Comment (nikipat) [reply content/calico/stars_policy_demo/create_resources.md]: will be pinging about this on Chime

Comment (dalbhanj) [reply content/calico/stars_policy_demo/create_resources.md]: I think you forgot to clone the source repo before this command
cd ~/environment/calico_resources
kubectl apply -f 00-namespace.yaml

**Comment by dalbhanj** [comment]
Hi @nikipat, everything looks good with just one minor change (see review). Please make changes and ping me when its done so we can merge this in

File: content/calico/stars_policy_demo/create_resources.md (hunk 2 above)
Comment (nikipat) [reply content/calico/stars_policy_demo/create_resources.md]: Hi Peter, I have cloned the repo in content/prerequisites/clone.md, under commit 891580f5723b88649d5bd4e769ae57f3e5a88876 (https://github.com/aws-samples/eks-workshop/pull/111/commits/891580f5723b88649d5bd4e769ae57f3e5a88876)

Comment (dalbhanj) [reply content/calico/stars_policy_demo/create_resources.md]: Could you move the cloning repo instructions into your chapter? this is mainly for keeping each chapter self-contained. You can add this right before creating namespace

Comment (nikipat) [reply content/calico/stars_policy_demo/create_resources.md]: I have moved it to my chapter, and committed my changes: e910c9d661f7a51431701018a1a026a2b97d33f7
